src/test/java/org/apache/commons/lang3/time/StopWatchTest.java

```diff
@@ -448,4 +463,292 @@ public void testToStringWithMessage() throws InterruptedException {
+            final StopWatch watch = StopWatch.create();
+
+            watch.start();
+            watch.suspend();
+
+            assertTrue(watch.isSuspended(), "Watch should be suspended");
+
+            Consumer<String> consumer = a -> {};
+
+            watch.accept(consumer).accept("a");
+
+            assertTrue(watch.isSuspended(), "Watch should be suspended");
+        }
+    }
+
+    @Nested
+    public class BiConsumerSupportRelatedTests {
+        @Test
+        public void testAccept() {
+            final StopWatch watch = StopWatch.create();
+
+            BiConsumer<String, String> consumer = (a, b) -> {};
+
+            watch.accept(consumer).accept("a", "b");
+
+            assertTrue(watch.isSuspended(), "Watch should be suspended");
+        }
+
+        @Test
+        public void testAcceptWhenStopWatchHasBeenSuspended() {
+            final StopWatch watch = StopWatch.create();
+
+            watch.start();
+            watch.suspend();
+
+            assertTrue(watch.isSuspended(), "Watch should be suspended");
+
+            BiConsumer<String, String> consumer = (a, b) -> {};
+
+            watch.accept(consumer).accept("a", "b");
+
+            assertTrue(watch.isSuspended(), "Watch should be suspended");
+        }
+    }
+
+    @Nested
+    public class TriConsumerSupportRelatedTests {
+        @Test
+        public void testAccept() {
+            final StopWatch watch = StopWatch.create();
+
+            TriConsumer<String, String, String> consumer = (a, b, c) -> {};
+
+            watch.accept(consumer).accept("a", "b", "c");
+
+            assertTrue(watch.isSuspended(), "Watch should be suspended");
+        }
+
+        @Test
+        public void testAcceptWhenStopWatchHasBeenSuspended() {
+            final StopWatch watch = StopWatch.create();
+
+            watch.start();
+            watch.suspend();
+
+            assertTrue(watch.isSuspended(), "Watch should be suspended");
+
+            TriConsumer<String, String, String> consumer = (a, b, c) -> {};
+
+            watch.accept(consumer).accept("a", "b", "c");
+
+            assertTrue(watch.isSuspended(), "Watch should be suspended");
+        }
+    }
+
+    @Nested
+    public class PredicateSupportRelatedTests {
+        @Test
+        public void testTest() {
+            final StopWatch watch = StopWatch.create();
+
+            Predicate<String> predicate = a -> true;
+
+            boolean result = watch.test(predicate).test("a");
+
+            assertTrue(result, "Result should be true");
+            assertTrue(watch.isSuspended(), "Watch should be suspended");
+        }
+
+        @Test
+        public void testTestWhenStopWatchHasBeenSuspended() {
+            final StopWatch watch = StopWatch.create();
+
+            watch.start();
+            watch.suspend();
+
+            assertTrue(watch.isSuspended(), "Watch should be suspended");
+
+            Predicate<String> predicate = a -> true;
```

Comment:
Also test using FailablePredicate.TRUE and other failable constants.

Reply:
Done. I use them if possible